src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -2042,7 +2211,149 @@ expr_ref seq_rewriter::is_nullable(expr* r) {
     return result;
 }
 
+br_status seq_rewriter::derivative(expr* hd, expr* r, expr_ref& result) {
```

Comment:
perhaps to avoid some clutter revert to signature

expr_ref seq_rewriter:;derivative(expr* elem, expr* r)

Have it reurn expr_ref(m()) (which is a null expression) on failure to rewrite.

Reply:
That makes sense, I was wondering if there was a convention for returning basically 'option expr'